### PR TITLE
feat: gql resolver for dataset examples

### LINKED ISSUE #3182: [datasets][gql] Examples connection

### PR DESCRIPTION
**Pull request by axiomofjoy**
resolves #3182

### CODE REVIEW

File: src/phoenix/server/api/types/Dataset.py [hunk 1]
```diff
@@ -59,3 +61,72 @@ async def versions(
...
+            .join(
+                latest_revisions_cte,
+                onclause=and_(
+                    (
+                        models.DatasetExampleRevision.dataset_example_id
+                        == latest_revisions_cte.c.dataset_example_id
+                    ),
+                    (
+                        models.DatasetExampleRevision.dataset_version_id
+                        == latest_revisions_cte.c.latest_dataset_version_id
```
Comment (RogerHYang) [inline src/phoenix/server/api/types/Dataset.py:106]: maybe too verbose? haha
```suggestion
                        == latest.c.version_id
```

**Review by RogerHYang — APPROVED** [review]

File: src/phoenix/server/api/types/Dataset.py [hunk 2]
```diff
@@ -59,3 +61,72 @@ async def versions(
...
+            )
+            .join(
+                models.DatasetExample,
+                onclause=models.DatasetExample.id
+                == models.DatasetExampleRevision.dataset_example_id,
+            )
+            .where(
+                models.DatasetExampleRevision.revision_kind != "DELETE",
+            )
+        )
```
[the remark this answers is not in the document]
Comment (axiomofjoy) [reply src/phoenix/server/api/types/Dataset.py]: Good idea, addressed in 986b0d84a9525ad5fc16a9a4f5b5dec5eb860e83.

File: src/phoenix/server/api/types/Dataset.py [hunk 3]
```diff
@@ -59,3 +61,72 @@ async def versions(
...
+        async with info.context.db() as session:
+            dataset_examples = [
+                DatasetExample(
+                    id_attr=dataset_example_revision.id,
+                    input=dataset_example_revision.input,
+                    output=dataset_example_revision.output,
+                    metadata=dataset_example_revision.metadata_,
+                    created_at=dataset_example_created_at,
+                )
+                for dataset_example_revision, dataset_example_created_at in await session.execute(
```
[the remark this answers is not in the document]
Comment (axiomofjoy) [reply src/phoenix/server/api/types/Dataset.py]: Nice, addressed in ac1da12d63e5d9bfab92c7fbe39cd5f51a0cdc67

File: src/phoenix/server/api/types/Dataset.py (hunk 1 above)
Comment (axiomofjoy) [reply src/phoenix/server/api/types/Dataset.py]: meet you halfway haha

addressed in 57bb281c33dd018010734ec3895648dd4b359371

**Pull request merged**